third_party/nvfuser/csrc/lower_utils.cpp

```diff
@@ -249,6 +249,13 @@ c10::optional<IterDomain*> getMaybeWarpReductionDim(
   }
 
   auto tv_in = getTv(input);
+
+  // __shfl_xor_sync() doesn't support complex number
+  if (tv_in->dtype() == DataType::ComplexFloat ||
+      tv_in->dtype() == DataType::ComplexDouble) {
+    return c10::nullopt;
+  }
+
```

Comment:
It is fine to do so in this PR, but in the future, we should consider using `__shfl_xor_sync` to exchange the real and imag part of the complex separately so we can do warp reduction on complex numbers.

Reply:
issue #2526 is added to to track this.